app/src/main/java/ch/epfl/skysync/components/forms/SpecialFields.kt

```diff
@@ -125,27 +127,32 @@ fun LocationPickerField(
     location: LocationPoint,
     defaultPadding: Dp,
     title: String,
-    setLocation: (LocationPoint) -> Unit
+    viewModel: FinishedFlightsViewModel,
+    setLocation: (LocationPoint) -> Unit,
 ) {
-  // TODO implement search bar logic
   var active by remember { mutableStateOf(false) }
   var query by remember { mutableStateOf("") }
+  val result by viewModel.searchResults.collectAsStateWithLifecycle()
```

Comment:
Nitpick: `result` is a list so you can add an 's' `results`

Reply:
Changed it